src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioMutation.cs

```diff
@@ -43,6 +46,42 @@ public void MutabilityTest()
             }
         }
 
+        [Fact]
+        public void MutabilityTest_Chain()
+        {
+            var config = new PowerFxConfig();            
+            config.AddFunction(new Assert2Function());
+            config.AddFunction(new Set2Function());
+            config.AddFunction(new Set3Function());
+            var engine = new RecalcEngine(config);            
+            var exprs = new string[]
+            {
+                "Assert2(obj.prop, 123); Set2(obj, \"prop\", 456); Assert2(obj.prop, 456)",
+                "Assert2(obj.prop, 123); Set3(obj, \"prop\", \"prop2\"); Assert2(obj.prop, 456)"
+            };
+
+            foreach (var expr in exprs)
+            {
+                var d = new Dictionary<string, FormulaValue>
+                {
+                    ["prop"] = FormulaValue.New(123),
+                    ["prop2"] = FormulaValue.New(456)
+                };
+
+                var obj = MutableObject.New(d);
+                engine.UpdateVariable("obj", obj);
+
+                var x = engine.Eval(expr, options: new ParserOptions() { AllowsSideEffects = true }); // Assert failures will throw.
+
+                if (x is ErrorValue ev)
+                {
+                    throw new XunitException($"FormulaValue is ErrorValue: {string.Join("\r\n", ev.Errors)}");
+                }
```

Comment:
this is asserting we're not a failure. 
The stronger assert is to assert we're returning the expected value. 

Although Assert2() is not an ideal function for this since it returns blank ;)

Maybe have that return something (even just 'true') and then we can verify.

Reply:
Done

Reply:
This is good. Now you can remove the (x is ErrorValue ev) check - the new NumberValue check is a stronger assert that supersedes it.